transforms/htmlTemplateCleanup/invalidTemplateAttributes.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { walkParse5Ast, serializeParse5Ast, deleteNode } from '../parse5Utils'
+import { TemplateDirectiveName, RootDirectiveName } from '@lwc/template-compiler'
+
+const rootTemplateDirectives = new Set(Object.values(RootDirectiveName))
+// TODO: @lwc/template-compiler doesn't include 'if:false in it's type definition
+// replace the 'if:false' once the type definition is updated
+const nonRootTemplateDirectives = new Set([...Object.values(TemplateDirectiveName), 'if:false'])
+
+const isValidRootTemplateDirective = (attr) => rootTemplateDirectives.has(attr.name)
+const isValidNonRootTemplateDirective = (attr) =>
+// Note @lwc/template-compiler does not expose an iterator type but rather refers to it as 'forof'
+  nonRootTemplateDirectives.has(attr.name) || attr.name.startsWith('iterator:')
+
+export const fixInvalidTemplateAttributes = ({ ast, file }, result) => {
+  let modifiedSource = false
+
+  walkParse5Ast(ast, (node) => {
+    const isRootTemplate = node.tagName === 'template' && node.parentNode === ast
```

Comment:
I wanted to detect the root node by using the `parentNode` but it looks like when the element is a `template`, `parse5` always sets `parentNode` to be a `#document-fragment` which does not have a parent.

For example 

```html
<template>
    <template if:true>
        ...
    </template>
</template>
```

In both cases the `template` node's `parentNode` is a `#document-fragment` which has no parent.

The only distinguishing feature between the two `#document-fragment` nodes is the root one has an `undefined` `sourceCodeLocation` while the non-root is defined.

I felt like checking for `sourceCodeLocation` is a bit fragile, especially since it's defined as an [optional value](https://github.com/inikulin/parse5/blob/master/packages/parse5/lib/tree-adapters/default.ts#L25), I'm not sure what other scenarios it will be missing.

Opted to just check that the`parentNode` matches the first AST node instead.

Reply:
`node.parentNode === ast` looks good to me!